R/html_dependency.R

```diff
@@ -231,32 +231,37 @@ copyDependencyToDir <- function(dependency, outputDir, mustWork = TRUE) {
     }
   }
 
-  if (!file.exists(outputDir))
+  if (!dir_exists(outputDir))
     dir.create(outputDir)
 
   target_dir <- file.path(outputDir,
     paste(dependency$name, dependency$version, sep = "-"))
 
-  if (!file.exists(target_dir)) {
-    dir.create(target_dir)
+  if (dir_exists(target_dir))
+    unlink(target_dir, recursive = TRUE)
```

Comment:
The reason for this was described in the commit message 480917c1989bc601e9e3f12676055d6fed8de30e Basically we don't there are leftover files in the target dependency dir, e.g. you may have lib/foo.js last time, and it was removed from the original library, then the next time you copy the library over to the target dir, you want to remove this lib/foo.js as well. It won't be removed only with `file.copy(recursive = TRUE)`.

I'll add a comment to explain it to the R code here.